### PR TITLE
MUL-151: Add support for inventory management by API

### PR DESCRIPTION
**Pull request by Wojdylak**

### CODE REVIEW

File: spec/Api/SectionResolver/ShopVendorApiUriBasedSectionResolverSpec.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        $this->shouldHaveType(UriBasedSectionResolverInterface::class);
+    }
+
+    public function it_returns_shop_vendor_section_if_path_starts_with_api_v2_shop_vendor(): void
+    {
+        $this->getSection('/api/v2/shop/account/vendor')->shouldBeLike(new ShopVendorApiSection());
+        $this->getSection('/api/v2/shop/account/vendor/something')->shouldBeLike(new ShopVendorApiSection());
+    }
+
+    public function it_throws_an_exception_if_path_does_not_start_with_api_v2_shop_vendor(): void
```
Comment (senghe) [inline spec/Api/SectionResolver/ShopVendorApiUriBasedSectionResolverSpec.php:38]: I think you should split this into separate specs

File: src/Api/Doctrine/ProductVariantCurrentVendorExtension.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+use BitBag\OpenMarketplace\Api\SectionResolver\ShopVendorApiSection;
+use BitBag\OpenMarketplace\Entity\VendorInterface;
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+final class ProductVariantCurrentVendorExtension implements QueryCollectionExtensionInterface, QueryItemExtensionInterface
+{
+    public function __construct(
+        private VendorContextInterface $vendorContext,
```
Comment (senghe) [inline src/Api/Doctrine/ProductVariantCurrentVendorExtension.php:27]: I think we should use older method for code-style consistency

File: src/Api/Doctrine/VendorAwareExtension.php [hunk 3]
```diff
@@ -15,17 +15,18 @@
...
 
 final class VendorAwareExtension implements ContextAwareQueryCollectionExtensionInterface, QueryItemExtensionInterface
 {
-    private VendorContextInterface $vendorContext;
-
-    public function __construct(VendorContextInterface $vendorContext)
-    {
-        $this->vendorContext = $vendorContext;
+    public function __construct(
+        private VendorContextInterface $vendorContext,
```
Comment (senghe) [inline src/Api/Doctrine/VendorAwareExtension.php:27]: Consistency first! :)

Comment (senghe) [reply src/Api/Doctrine/VendorAwareExtension.php:27]: And it probably will be removed :upside_down_face:

File: src/Api/SectionResolver/ShopVendorApiUriBasedSectionResolver.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace BitBag\OpenMarketplace\Api\SectionResolver;
+
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionCannotBeResolvedException;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionInterface;
+use Sylius\Bundle\CoreBundle\SectionResolver\UriBasedSectionResolverInterface;
+
+final class ShopVendorApiUriBasedSectionResolver implements UriBasedSectionResolverInterface
+{
+    public function __construct(private string $shopVendorApiUriBeginning)
```
Comment (senghe) [inline src/Api/SectionResolver/ShopVendorApiUriBasedSectionResolver.php:19]: Consistency :)

File: src/Api/SectionResolver/ShopVendorApiUriBasedSectionResolver.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    {
+    }
+
+    public function getSection(string $uri): SectionInterface
+    {
+        if (!str_starts_with($uri, $this->shopVendorApiUriBeginning)) {
+            throw new SectionCannotBeResolvedException();
+        }
+
+        return new ShopVendorApiSection();
```
Comment (senghe) [inline src/Api/SectionResolver/ShopVendorApiUriBasedSectionResolver.php:29]: We should use factory here

File: src/Api/Serializer/ProductVariantNormalizer.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+use BitBag\OpenMarketplace\Api\SectionResolver\ShopVendorApiSection;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+final class ProductVariantNormalizer implements ContextAwareNormalizerInterface, NormalizerAwareInterface
+{
+    public function __construct(
+        private ContextAwareNormalizerInterface $productVariantNormalizer,
```
Comment (senghe) [inline src/Api/Serializer/ProductVariantNormalizer.php:22]: Could you check this globally? :thinking:

File: src/Resources/config/serialization/ProductVariant.xml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+This file was created by developers working at BitBag
+Do you need more information about us and what we do? Visit our https://bitbag.io website!
+We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+-->
+
+<serializer xmlns="http://symfony.com/schema/dic/serializer-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/serializer-mapping https://symfony.com/schema/dic/serializer-mapping/serializer-mapping-1.0.xsd"
+>
+    <class name="Sylius\Component\Core\Model\ProductVariant">
```
Comment (senghe) [inline src/Resources/config/serialization/ProductVariant.xml:13]: How about using parameter instead of hardcoded name?

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Api/Doctrine/VendorAwareExtension.php (hunk 3 above)
Comment (Wojdylak) [reply src/Api/Doctrine/VendorAwareExtension.php:27]: In the next PR

File: src/Resources/config/serialization/ProductVariant.xml (hunk 7 above)
Comment (Wojdylak) [reply src/Resources/config/serialization/ProductVariant.xml:13]: I checked. When we use parameter, the object does not serialize

File: src/Factory/ShopVendorApiSectionFactoryInterface.php [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
```
Comment (senghe) [inline src/Factory/ShopVendorApiSectionFactoryInterface.php:3]: No OS header :)

File: src/Factory/ShopVendorApiSectionFactoryInterface.php [hunk 9]
```diff
@@ -0,0 +1,13 @@
+<?php
```
Comment (senghe) [inline src/Factory/ShopVendorApiSectionFactoryInterface.php:1]:
```suggestion
<?php


/*
 * This file was created by developers working at BitBag
 * Do you need more information about us and what we do? Visit our https://bitbag.io website!
 * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
*/
```

**Review by senghe — APPROVED** [review]

**Pull request merged**